test/model/compose_test.dart

```diff
@@ -0,0 +1,228 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/model/compose.dart';
+
+void main() {
+  group('wrapWithBacktickFence', () {
+    /// Check `wrapWithBacktickFence` on example input and expected output.
+    ///
+    /// The intended input (content passed to `wrapWithBacktickFence`)
+    /// is straightforward to infer from `expected`.
+    /// To do that, this helper takes `expected` and removes the opening and
+    /// closing fences.
+    ///
+    /// Then we have the input to the test, as well as the expected output.
+    void checkFenceWrap(String expected, {String? infoString, bool chopNewline = false}) {
+      final re = RegExp(r'^.*?\n(.*\n).*\n$', dotAll: true);
+      String content = re.firstMatch(expected)![1]!;
+      if (chopNewline) content = content.substring(0, content.length - 1);
+      check(wrapWithBacktickFence(content: content, infoString: infoString)).equals(expected);
+    }
+
+    test('empty content', () {
+      checkFenceWrap(chopNewline: true, '''
+```
+
+```
+''');
```

Comment:
Hmmm, this example doesn't look right. The output is a block whose contents are one empty line, not a block with empty contents.

I think the trouble is that that condition I suggested above:
```dart
  if (!content.endsWith('\n')) {
    resultBuffer.write('\n');
```

is incorrectly treating an empty string as ending with an incomplete line. So it should instead be like `if (content.isNotEmpty && !content.endsWith('\n'))`.